crates/ruff/src/importer/mod.rs

```diff
@@ -34,13 +40,19 @@ impl<'a> Importer<'a> {
             python_ast,
             locator,
             stylist,
-            ordered_imports: Vec::default(),
+            runtime_imports: Vec::default(),
+            type_checking_blocks: Vec::default(),
         }
     }
 
     /// Visit a top-level import statement.
     pub(crate) fn visit_import(&mut self, import: &'a Stmt) {
-        self.ordered_imports.push(import);
+        self.runtime_imports.push(import);
+    }
+
+    /// Visit a top-level type-checking block.
+    pub(crate) fn visit_type_checking_block(&mut self, type_checking_block: &'a Stmt) {
```

Comment:
Nit: Change to `StmtIf` because that's the only block that we support?

```suggestion
    pub(crate) fn visit_type_checking_block(&mut self, type_checking_block: &'a StmtIf) {
```